storage/innobase/row/row0ins.cc

```diff
@@ -1755,18 +1755,57 @@ row_ins_check_foreign_constraint(
 		cmp = cmp_dtuple_rec(entry, rec, offsets);
 
 		if (cmp == 0) {
-			if (rec_get_deleted_flag(rec,
-						 rec_offs_comp(offsets))) {
-				err = row_ins_set_shared_rec_lock(
-					LOCK_ORDINARY, block,
-					rec, check_index, offsets, thr);
-				switch (err) {
-				case DB_SUCCESS_LOCKED_REC:
-				case DB_SUCCESS:
-					break;
-				default:
-					goto end_scan;
-				}
+                  if (DICT_TF2_FLAG_IS_SET(check_index->table,
+                                           DICT_TF2_VERSIONED)) {
+                    mem_heap_t *clust_heap = mem_heap_create(256);
+                    ulint clust_offsets_[REC_OFFS_NORMAL_SIZE];
+                    ulint *clust_offsets = clust_offsets_;
+                    rec_offs_init(clust_offsets_);
+                    btr_pcur_t clust_pcur;
+
+                    ut_a(!dict_index_is_clust(check_index));
```

Comment:
Why `check_index` can't be clustered index? I think it should be main use case. Or I missed something...